test/unit/view_model/active_record/cache_test.rb

```diff
@@ -0,0 +1,276 @@
+
+DUMMY_RAILS_CACHE = ActiveSupport::Cache::MemoryStore.new
+
+module Rails
+  def self.cache
+    DUMMY_RAILS_CACHE
+  end
+end
+
+class ViewModel::ActiveRecord
+  class CacheTest < ActiveSupport::TestCase
+    using ViewModel::Utils::Collections
+    extend Minitest::Spec::DSL
+
+    # Defines a cacheable parent Model with a owned Child and a cachable shared Shared.
+    module CacheableParentAndChildren
+      extend ActiveSupport::Concern
+      include ViewModelSpecHelpers::ParentAndBelongsToChild
+
+      def model_attributes
+        super.merge(
+          schema:    ->(t) { t.references :shared, foreign_key: true },
+          model:     ->(_) { belongs_to :shared, inverse_of: :models },
+          viewmodel: ->(_) {
+            association :shared, shared: true, optional: false
+            cacheable!
+          }
+        )
+      end
+
+      def shared_viewmodel_class
+        @shared_viewmodel_class ||= define_viewmodel_class(:Shared, namespace: namespace) do
+          define_schema do |t|
+            t.string :name
+          end
+
+          define_model do
+            has_many :models
+          end
+
+          define_viewmodel do
+            attributes :name
+            cacheable!
+          end
+        end
+      end
+
+      def shared_model_class
+        shared_viewmodel_class.model_class
+      end
+
+      # parent depends on children, ensure it's touched first
+      def viewmodel_class
+        shared_viewmodel_class
+        super
+      end
+
+      included do
+        let(:shared)     { shared_model_class.create!(name: "shared1") }
+        let(:root)       { model_class.create!(name: "root1", child: Child.new(name: "owned1"), shared: shared) }
+        let(:root_view)  { viewmodel_class.new(root) }
+      end
+    end
+
+    before(:each) do
+      DUMMY_RAILS_CACHE.clear
+    end
+
+    # Extract the iKnowCaches to verify their contents
+    def cache_for(viewmodel)
+      viewmodel.viewmodel_cache.send(:cache_specialization_for, viewmodel.new_serialize_context)
+    end
+
+    def serialize_from_database
+      view    = viewmodel_class.new(model_class.find(root.id))
+      context = viewmodel_class.new_serialize_context
+      data    = ViewModel.serialize_to_hash([view], serialize_context: context)
+      refs    = context.serialize_references_to_hash
+      [data, refs]
+    end
+
+    def parse_result(result)
+      data_json, refs_json = result
+      data                 = data_json.map { |d| JSON.parse(d) }
+      refs                 = refs_json.transform_values { |v| JSON.parse(v) }
+      [data, refs]
+    end
+
+    def fetch_with_cache
+      viewmodel_class.viewmodel_cache.fetch([root.id])
+    end
+
+    def serialize_with_cache
+      parse_result(fetch_with_cache)
+    end
+
+    module BehavesLikeACache
+      extend ActiveSupport::Concern
+      included do
+        it 'returns the right serialization' do
+          (serialize_with_cache).must_equal(serialize_from_database)
```

Comment:
Stray brackets? This pattern occurs a few times, and I'm not familiar with it. What's the significance?

Reply:
I mechanically converted from `expect(x).to equal(y)`. Converting and dropping the brackets looked like it lost something to me; the emphasis on the expressions that were being compared vanished. So I left them in. It is weird though. Do you think it adds enough to keep?